### DISCUSSION TITLE
Set additional flags for kafka_exporter

### DISCUSSION THREAD

**Discussion by davidpiendl** [strimzi/strimzi-kafka-operator#7233] [Q&A] [answered]
Hi,
is it somehow easy possible to configure [additional properties](https://github.com/danielqsj/kafka_exporter#flags) of the kafka_exporter? In the CRD there are only a few available (i.e groupRegex, topicRegex ...).

Best regards

**Comment by scholzj** [comment, marked as answer]
No, I don't think that is possible. Not without modyfing the container image and the startup scripts inside. Why? What would you want to configure?

**Comment by davidpiendl** [comment]
We want to test a [new setting](https://github.com/danielqsj/kafka_exporter/commit/5cb32d4bfeb48636ef1b5d5ebbb5856c6bc776d4) for the used sarama client in the exporter. We run our cluster with enabled autocreation of topics. For consumers and metadata fetches we hope that this setting will fix our experience with recreated topics after deletion with no explicit producer/consumer running. This is most likely caused by the exporter.

**Reply by scholzj** [reply]
The Exporter does not read your topics. So I do not think it will recreate them. I would expect it to apply to the topics it reads such as `__consumer_offsets`. But it is not completely clear. I did not found any PR or Issue for this flag.

**Reply by davidpiendl** [reply]
The original issue is this one https://github.com/danielqsj/kafka_exporter/issues/253 there is also another one having the same issue https://github.com/danielqsj/kafka_exporter/issues/45. We also experience the same behaviour with strimzi operator 0.28 - kafka 3.0.0 and exporter 1.4.2

**Reply by scholzj** [reply]
Hmm, interesting. But it should not be consuming or producing messages to your topics, so why would it auto-create them? In any case, Strimzi does not currently use 1.5.0 as it was released few hours ago only (but I will get it updated). The option is not configurable either. So you would need to modify the image and its startup script.

**Reply by davidpiendl** [reply]
Thanks for your quick responses. As far as I've read, simple metadata requests are also enough to trigger topic creation.

**Reply by scholzj** [reply]
So, do you know what topics does it auto-create? The flag defaults to `true` -> I really wonder why does it actually want to default to true. What topics does it auto-create?

**Reply by scholzj** [reply]
> Thanks for your quick responses. As far as I've read, simple metadata requests are also enough to trigger topic creation.

In that case, the topic auto-creation is even much more evil then I expected :-o

**Reply by scholzj** [reply]
So, apparently you are right, a metadata request triggers it. I guess you present a good case why it should be disabled. It would be good to understand what would be the case to have it enabled. If there is one, a new option can be added to disable it. If there isn't one, we can just disable it by default in the container image without changing the API which would be easier.

**Reply by davidpiendl** [reply]
We enabled it due to the fact we have an asynchronous environment where it can be the case, that topics are deleted, but there are still events in our producing streaming pipeline, which require the topic to exist. Any (Admin)Api call in this streaming pipeline to check if topics exist, or even always trying to create the topic explicitely (even if it still exists), would impact our performance. Therefore we rely on the autocreation feature, but sadly this feature does to much :(.
